util/test/Test/Pos/Util/Tripping.hs

```diff
@@ -1,34 +1,52 @@
-{-# OPTIONS_GHC -fno-warn-orphans #-}
+{-# OPTIONS_GHC -fno-warn-orphans      #-}
+{-# OPTIONS_GHC -fno-warn-deprecations #-}
 
 module Test.Pos.Util.Tripping where
 
 import qualified Prelude
 import           Universum
 
-import           Data.Aeson (FromJSON, ToJSON, decode, encode)
+import           Data.Aeson (FromJSON, ToJSON, eitherDecode, encode)
 import           Data.Text.Internal.Builder (fromText, toLazyText)
+import qualified Data.Yaml as Y
 import           Formatting.Buildable (Buildable (..))
-import           Hedgehog (Group, MonadTest, discoverPrefix, success, tripping)
-import           Hedgehog.Internal.Property (Diff (..), failWith)
-import           Hedgehog.Internal.Show (valueDiff)
+import           Hedgehog (Gen, Group, MonadTest, discoverPrefix, success, tripping)
+import           Hedgehog.Internal.Property (Diff (..), Property, TestLimit, annotate,
+                     failWith, forAllWith, property, withTests)
+import           Hedgehog.Internal.Show (showPretty, valueDiff)
 import           Hedgehog.Internal.TH (TExpQ)
 import           System.IO (hSetEncoding, stderr, stdout, utf8)
 import qualified Text.JSON.Canonical as Canonical
 import           Text.Show.Pretty (Value (..), parseValue)
 
 import           Pos.Util.Json.Canonical (SchemaError (..))
 
+roundTripsAesonYamlShow :: (Eq a, Show a, ToJSON a, FromJSON a) => TestLimit -> Gen a -> Property
+roundTripsAesonYamlShow testLimit things = withTests testLimit . property $ do
+     annotate "Aeson"
+     forAllWith
+         (\x -> yamlAesonCustomRender x encode eitherDecode) things >>= roundTripsAesonShow
+     annotate "Yaml"
+     forAllWith
+         (\x -> yamlAesonCustomRender x Y.encode Y.decodeEither') things >>= roundTripsYAMLShow
+
 discoverRoundTrip :: TExpQ Group
 discoverRoundTrip = discoverPrefix "roundTrip"
 
 roundTripsAesonShow
+    :: (Eq a, MonadTest m, ToJSON a, FromJSON a, Show a, HasCallStack) => a -> m ()
+roundTripsAesonShow a = tripping a encode eitherDecode
+
+roundTripsYAMLShow
     :: (Eq a, MonadTest m, ToJSON a, FromJSON a, Show a) => a -> m ()
-roundTripsAesonShow a = tripping a encode decode
+roundTripsYAMLShow a = tripping a Y.encode Y.decodeEither
```

Comment:
Because there is no easily derivable `Eq` instance for `Y.decodeEither'` (specifically `ParseException` http://hackage.haskell.org/package/yaml-0.11.0.0/docs/Data-Yaml.html#t:ParseException) I used `Y.decodeEither` but this function is deprecated. Any feedback welcomed wrt alternative solutions.

Reply:
This patch should do what I'm suggesting:
```
diff --git a/util/test/Test/Pos/Util/Tripping.hs b/util/test/Test/Pos/Util/Tripping.hs
index 74ef96960..1c902583b 100644
--- a/util/test/Test/Pos/Util/Tripping.hs
+++ b/util/test/Test/Pos/Util/Tripping.hs
@@ -39,19 +39,22 @@ roundTripsAesonShow a = tripping a encode eitherDecode
 
 roundTripsYAMLShow
     :: (Eq a, MonadTest m, ToJSON a, FromJSON a, Show a) => a -> m ()
-roundTripsYAMLShow a = tripping a Y.encode Y.decodeEither
+roundTripsYAMLShow a = tripping a Y.encode Y.decodeEither'
 
 -- | Round trip any `a` with both `ToJSON` and `FromJSON` instances
 roundTripsAesonYamlBuildable
     :: (Eq a, MonadTest m, ToJSON a, FromJSON a, Buildable a) => a -> m ()
 roundTripsAesonYamlBuildable a = do
     trippingBuildable a encode eitherDecode "Aeson"
-    trippingBuildable a Y.encode Y.decodeEither "Yaml"
+    trippingBuildable a Y.encode Y.decodeEither' "Yaml"
 
 -- We want @SchemaError@s to show up different (register failure)
 instance Eq SchemaError where
     _ == _ = False
 
+instance Eq Y.ParseException where
+    _ == _ = False
+
 roundTripsCanonicalJSONShow
     :: forall m a
      . ( Eq a
@@ -130,6 +133,10 @@ instance Buildable a => Buildable (Either String a)  where
     build (Left l) = build l
     build (Right r) = build r
 
+instance Buildable a => Buildable (Either Y.ParseException a)  where
+    build (Left l) = fromText (show l)
+    build (Right r) = build r
+
 instance Buildable () where
     build () = "()"
 ```